src/Web3Provider.jsx

```diff
@@ -103,12 +104,14 @@ class Web3Provider extends React.Component {
 
     if (isEmpty(ethAccounts)) {
       web3 && web3.currentProvider && web3.currentProvider.enable()
```

Comment:
@coopermaruyama There seems to be some nuance with using `currentProvider.enable()` vs `eth.getAccounts()`. 

**Using currentProvider.enable():**
If metamask in installed and user is not logged in, rather than rendering `AccountUnavailable`, metamask will popup with a login screen.

**Using eth.getAccounts():**
If metamask in installed and user is not logged in, we simply render `AccountUnavailable`.

I went ahead and kept the usage of `currentProvider.enable()` since a previous PR implemented it for EIP-1102. I'll let you make the call here.

Reply:
@gohuygo What if something in `this.props.children` has a dependency on the availability of accounts (e.g. throws an exception if no account exists)?

Reply:
Good point. It looks like my flag is completely unnecessary here since we call `getAccounts()` in the constructor. This flickering issue looks like it was introduced with EIP-1102's `currentProvider.enable()` since reverting it fixed the issue. I'll have to dig into this some more. Closing this PR